src/components/PatentVisualizer/PatentVisualizer.js

```diff
@@ -1,193 +1,207 @@
-                        <Button type="primary">Edit Data</Button>
-                    </div>
-                </Sider>
-                }
-            </Layout>,
-            <PatentTable />
-        ]
-    )
-};
-
+import React, { useState, useEffect, useRef } from 'react';
+import 'antd/dist/antd.css';
+import './PatentVisualizer.css';
+import { CloseOutlined } from '@ant-design/icons';
+import { Heatmap } from '@ant-design/charts';
+import { Layout, Button } from 'antd';
+import PatentVisualizerSidebar from './PatentVisualizerSidebar';
+import PatentTable from './PatentTable';
+import { assignColors } from '../../utils/colors';
+import { getUnique } from '../../utils/utils';
+import { getPatentData, generateVisualizationDataset } from '../../utils/patentDataUtils';
+import { useLocation } from 'react-router-dom';
+
+const { Sider } = Layout;
+
+const KEYS = {
+    assignee: 'patentAssignees',
+    sequencePosition: 'sequencePosition',
+    patentNumber: 'patentNumber',
+    claimed: 'Claimed',
+    aminoAcid: 'Amino Acid',
+    baseline: 'Sequence'
+}
+Object.freeze(KEYS);
+
+const getMaximumSeq = (patentArray) => {
+    return patentArray.reduce((max, current) => {
+        const seqPosition = parseInt(current[KEYS.sequencePosition]);
+        if(seqPosition > max) {
+            return seqPosition;
+        } else {
+            return max;
+        }
+    }, 0);
+}
+
+const PatentVisualizer = props => {
+    const location = useLocation();
+
+    const [data, setData] = useState([]);
+    const [colorKeys, setColorKeys] = useState({});
+    const [details, setDetails] = useState({ show: false, patentId: 0, seqPosition: 0 });
+    const [showBaseline, setBaseline] = useState(false);
+    const _dataRef = useRef([]);
+
+    useEffect(() => {
+        const proteinName = location.state.proteinName;
+        console.log('Fetching patent details for', proteinName)
+        const patentData = getPatentData(proteinName);
+        setPatentData(patentData);
+    }, []);
+
+    // Filtering Effect
+    const [assignees, setAssignees] = useState({});
+    const [sequenceRange, setSequenceRange] = useState({ min: 1, max: 1, length: 1 });
+    useEffect(() => {
+        setData(_dataRef.current
+            // By Assignee
+            .filter(patentData => {
+                // We remove the baseline sequence of this filter as it is not really an assignee but we need to format data that way
+                return assignees[patentData[KEYS.assignee]] || patentData[KEYS.assignee] === KEYS.baseline;
+            })
+            // By Sequence Range
+            .filter((patentData) => {
+                return sequenceRange.min <= patentData[KEYS.sequencePosition] && patentData[KEYS.sequencePosition] <= sequenceRange.max;
+            })
+        );
+    }, [assignees, sequenceRange, showBaseline]);
+
+    const setPatentData = (patentData) => {
+        Promise.resolve(patentData)

```

Comment:
Promise.resolve was part of the mock function. It looks like we have patentData already passed as argument here so you can just remove all the `Promise.resolve` `then` `catch`